### PR TITLE
[IMP] project: display the project name in another way in project sharing

### PR DESCRIPTION
**Pull request by xavierbol**
Before this commit, the action to display the project sharing feature when the project is shared in edit mode was given as an object and call by the JS code. The problem is we no longer used the `doAction` function in the action service and so the markup is not automatically given to the helper of the action and we have to add it to be sure the helper is rendered in HTML.

This commit replaces the action object given by the action name as it was before and keep the Project name in the breadcrumb when the portal user selects the project in his portal to access it in edit mode.

task-3615844

### CODE REVIEW

File: addons/project/views/project_sharing_project_task_views.xml [hunk 1]
```diff
@@ -294,6 +294,11 @@
         <field name="view_mode">kanban,tree,form</field>
         <field name="search_view_id" ref="project.project_sharing_project_task_view_search"/>
         <field name="domain">[('project_id', '=', active_id), ('display_in_project', '=', True)]</field>
+        <field name="context">{
+            'default_project_id': active_id,
+            'active_id_chatter': active_id,
+            'delete': false,
+        }</field>
```
Comment (pkyriakou) [inline addons/project/views/project_sharing_project_task_views.xml:301]: Not something that I think should change (just curious) but why is this needed here ? :thinking:

File: addons/sale_project/views/project_sharing_views.xml [hunk 2]
```diff
@@ -41,4 +41,13 @@
...
     </record>
 
+    <record id="project.project_sharing_project_task_action" model="ir.actions.act_window">
+        <field name="context">{
+            'default_project_id': active_id,
+            'active_id_chatter': active_id,
+            'delete': false,
+            'sale_show_partner_name': true,
+        }</field>
+    </record>
```
Comment (pkyriakou) [inline addons/sale_project/views/project_sharing_views.xml:51]: Same question for this one as well :thinking:

**Review by pkyriakou — COMMENTED** [review]
Thank you @xavierbol :100: :1st_place_medal: 
I left two questions but I will be greening the task regardless :green_circle: :v:

File: addons/project/views/project_sharing_project_task_views.xml (hunk 1 above)
Comment (xavierbol) [reply addons/project/views/project_sharing_project_task_views.xml:301]: Before my changes, the additional context was given in the python method (see [here](https://github.com/odoo/odoo/blob/b2c33e01f3fdd7ecda891b804af8cc9b4572933f/addons/project/models/project_project.py#L641-L646)), but since I remove that method, I have to pass the context in that action to keep the context we got before. 🙂 

(Also, I see I forgot to add `'search_default_open_tasks': True,` in the context, I will addd it)

**Review by auon-odoo — APPROVED** [review]

**Comment by auon-odoo** [comment]
@robodoo r+

**Comment by xavierbol** [comment]
robodoo rebase-ff
